### PR TITLE
[Metricbeat] etcd: fix leader metricset fields

### PR DESCRIPTION
**Pull request by ghost**
## What does this PR do?
Fixes etcd leader metricset fields.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by rdner** [comment]
@mergifyio update

File: metricbeat/module/etcd/leader/leader_integration_test.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ package leader
 import (
 	"testing"
 
-	"github.com/stretchr/testify/assert"
+	// "github.com/stretchr/testify/assert"
```
Comment (rdner) [inline metricbeat/module/etcd/leader/leader_integration_test.go:26]: Why is this and assert calls commented in this file?

File: metricbeat/module/etcd/leader/leader_test.go [hunk 2]
```diff
@@ -31,15 +31,6 @@ import (
...
 
-func TestEventMapping(t *testing.T) {
-	content, err := ioutil.ReadFile("../_meta/test/leaderstats.json")
-	assert.NoError(t, err)
-
-	event := eventMapping(content)
-
-	assert.Equal(t, event["leader"], string("924e2e83e93f2560"))
-}
-
```
Comment (rdner) [inline metricbeat/module/etcd/leader/leader_test.go:42]: We don't need this test anymore?

File: metricbeat/module/etcd/test_etcd.py [hunk 3]
```diff
@@ -11,7 +11,7 @@ class Test(metricbeat.BaseTest):
     COMPOSE_SERVICES = ['etcd']
 
     @parameterized.expand([
-        "leader",
+        # "leader",
```
Comment (rdner) [inline metricbeat/module/etcd/test_etcd.py:14]: Why is this commented?

File: metricbeat/module/etcd/leader/leader_integration_test.go (hunk 1 above)
Comment (gpop63) [reply metricbeat/module/etcd/leader/leader_integration_test.go:26]: The change requested to the `leader` metricset makes it that `leader` events are triggered only if the `etcd` instance has more than 1 follower. I had a chat with @sayden and we agreed that tests can be removed for this special case.

Comment (rdner) [reply metricbeat/module/etcd/leader/leader_integration_test.go:26]: Then let's remove it. Having this commented in the code is very confusing.

Comment (gpop63) [reply metricbeat/module/etcd/leader/leader_integration_test.go:26]: Remove the commented lines or the integration tests?

Comment (rdner) [reply metricbeat/module/etcd/leader/leader_integration_test.go:26]: @gpop63 My comment is about commented lines in this particular file. Not sure why we're talking about integration tests, sorry. Perhaps you replied to a wrong comment here?

File: metricbeat/module/etcd/leader/leader_test.go (hunk 2 above)
Comment (gpop63) [reply metricbeat/module/etcd/leader/leader_test.go:42]: `eventMapping` function has changed - it loops over etcd followers and doesn't return anything.

File: metricbeat/module/etcd/test_etcd.py (hunk 3 above)
Comment (gpop63) [reply metricbeat/module/etcd/test_etcd.py:14]: `leader` metricset events are triggered if the etcd instance has more than 0 followers - the testsuite starts etcd with 0 followers therefore no events will be found and tests will fail.

File: metricbeat/module/etcd/leader/leader_test.go (hunk 2 above)
Comment (rdner) [reply metricbeat/module/etcd/leader/leader_test.go:42]: @gpop63 but then I cannot see how we test the new behaviour now, I might be missing something.

Comment (rdner) [reply metricbeat/module/etcd/leader/leader_test.go:42]: Thank you very much @gpop63 !

**Review by rdner — APPROVED** [review]

**Pull request merged**